proto/prost/private/protoc_wrapper.rs

```diff
@@ -0,0 +1,1047 @@
+    protoc: PathBuf,
+
+    /// The path to the output directory.
+    out_dir: PathBuf,
+
+    /// The name of the crate.
+    crate_name: String,
+
+    /// The bazel label.
+    label: String,
+
+    /// The path to the package info file.
+    package_info_file: PathBuf,
+
+    /// The proto files to compile.
+    proto_files: Vec<PathBuf>,
+
+    /// The include directories.
+    includes: Vec<String>,
+
+    /// Dependency descriptor sets.
+    descriptor_set: PathBuf,
+
+    /// The path to the generated lib.rs file.
+    out_librs: PathBuf,
+
+    /// The proto include paths.
+    proto_paths: Vec<String>,
+
+    /// The path to the rustfmt binary.
+    rustfmt: Option<PathBuf>,
+
+    /// Whether to generate tonic code.
+    is_tonic: bool,
+
+    /// Extra arguments to pass to protoc.
+    extra_args: Vec<String>,
+}
+
+impl Args {
+    /// Parse the command-line arguments.
+    fn parse() -> Result<Args, String> {
+        let mut protoc: Option<PathBuf> = None;
+        let mut out_dir: Option<PathBuf> = None;
+        let mut crate_name: Option<String> = None;
+        let mut package_info_file: Option<PathBuf> = None;
+        let mut proto_files: Vec<PathBuf> = Vec::new();
+        let mut includes = Vec::new();
+        let mut descriptor_set = None;
+        let mut out_librs: Option<PathBuf> = None;
+        let mut rustfmt: Option<PathBuf> = None;
+        let mut proto_paths = Vec::new();
+        let mut label: Option<String> = None;
+        let mut tonic_or_prost_opts = Vec::new();
+        let mut is_tonic = false;
+
+        let mut extra_args = Vec::new();
+
+        // Iterate over the given command line arguments parsing out arguments
+        // for the process runner and arguments for protoc and potentially spawn
+        // additional arguments needed by prost.
+        for arg in env::args().skip(1) {
+            if !arg.starts_with('-') {
+                proto_files.push(PathBuf::from(arg));
+                continue;
+            }
+
+            if arg.starts_with("-I") {
+                includes.push(
+                    arg.strip_prefix("-I")
+                        .expect("Failed to strip -I")
+                        .to_string(),
+                );
+                continue;
+            }
+
+            if arg == "--is_tonic" {
+                is_tonic = true;
+                continue;
+            }
+
+            if !arg.contains('=') {
+                extra_args.push(arg);
+                continue;
+            }
+
+            let part = arg.split_once('=').expect("Failed to split argument on =");
+            match part {
+                ("--protoc", value) => {
+                    protoc = Some(PathBuf::from(value));
+                }
+                ("--prost_out", value) => {
+                    out_dir = Some(PathBuf::from(value));
+                }
+                ("--tonic_out", value) => {
+                    out_dir = Some(PathBuf::from(value));
+                }
```

Comment:
Let's assert that at most one of these are set, or collapse this back to only one arg that's used for both. (Or alternatively, save these as independent fields and use them to set the different flags potentially differently, or assert that `tonic_out` is only set if `is_tonic` and assert that the values are the same)

Reply:
`--tonic_out` is actually never passed through, so I'll just delete this block. Good catch.